src/compiler/checker.ts

```diff
@@ -1338,7 +1338,7 @@ namespace ts {
 
         function checkAndReportErrorForUsingTypeAsNamespace(errorLocation: Node, name: __String, meaning: SymbolFlags): boolean {
             if (meaning === SymbolFlags.Namespace) {
-                const symbol = resolveSymbol(resolveName(errorLocation, name, SymbolFlags.Type & ~SymbolFlags.Value, /*nameNotFoundMessage*/undefined, /*nameArg*/ undefined, /*isUse*/ false));
+                const symbol = resolveSymbol(resolveName(errorLocation, name, SymbolFlags.Class | SymbolFlags.Interface | SymbolFlags.TypeAlias, /*nameNotFoundMessage*/undefined, /*nameArg*/ undefined, /*isUse*/ false));

```

Comment:
It seems like type aliases might already have been resolved by this point, so I'm not sure the flag makes any difference.

Reply:
What about type parameters etc?

Reply:
I was (incorrectly) thinking of a nominal type system in which it would be preferable to just use the name of the base type to get at the property.

Reply:
Can you please elaborate on "etc"?  Are there other values that make sense?

Reply:
Offline, we settled on `SymbolFlags.Type & ~SymbolFlags.Namespace`.